apps/web/app/.server/service/ethereum-l1/contracts/guardians.ts

```diff
@@ -0,0 +1,50 @@
+import { getContract, type Address, type Hex } from "viem";
+
+import { l1Rpc } from "../client";
+import { guardiansAddress } from "./protocol-upgrade-handler";
+import { guardiansAbi } from "@/utils/contract-abis";
+
+const guardians = (address: Address) =>
+  getContract({
+    address,
+    abi: guardiansAbi,
+    client: l1Rpc,
+  });
+
+const GUARDIAN_MEMBERS_COUNT = 8;
+
+export async function withGuardiansAddress<T, A extends unknown[]>(
```

Comment:
I don't love the with... pattern. I would understand if it's something with a tear down, like open and closing a collection. But in this case, I believe that it's clearer just get the address.